### PR TITLE
FF7: Fix a softlock when trying to change in-game Controller setting to "Normal"

### PR DESCRIPTION
**Pull request by maciej-trebacz**
## Summary

Fixes issue #623 by disabling the ability to change the Controller setting to "Normal". This is what Aali driver originally did and this change apparently wasn't ported to FFNx. 

### ACKs

- [x] I have updated the [Changelog.md](https://github.com/julianxhokaxhiu/FFNx/blob/master/Changelog.md) file
- [x] I did test my code on FF7
- [ ] I did test my code on FF8

### CODE REVIEW

File: src/ff7_opengl.cpp [hunk 1]
```diff
@@ -84,6 +84,9 @@ void ff7_init_hooks(struct game_obj *_game_object)
 	// Allow mouse cursor to be shown
 	replace_function(ff7_externals.dinput_createdevice_mouse, noop);
 
+  // Disable "Normal" setting in Controller section of the Config menu (it softlocks on Steam)
+  memset_code(ff7_externals.config_menu_sub + 0x8AC, 0x90, 0xE6);
```
Comment (julianxhokaxhiu) [inline src/ff7_opengl.cpp:88]: Same as the other PR, please move these lines where the steam edition is being checked. Thank you

**Review by julianxhokaxhiu — CHANGES_REQUESTED** [review]
Thanks for this PR as well. Also here I left a comment on the specific part which would also impact 1998 edition.

File: src/ff7_opengl.cpp (hunk 1 above)
Comment (maciej-trebacz) [reply src/ff7_opengl.cpp]: @julianxhokaxhiu ok, moved the patch to the section that checks for Steam version.

**Review by julianxhokaxhiu — APPROVED** [review]

File: Changelog.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 - 60FPS: Refactor 60FPS battle menu fix and fix ESUI compatibility ( https://github.com/julianxhokaxhiu/FFNx/pull/597 )
 - Ambient: Allow ambient effects to playback in fields that use movies as background
+- Common: Fix a softlock when trying to change in-game Controller setting to "Normal" (by disabling this option)
```
Comment (julianxhokaxhiu) [inline Changelog.md:15]: Anything that is related to Steam used to have its own section on the Changelog, can you please move this line into the dedicated section? See https://github.com/julianxhokaxhiu/FFNx/blob/master/Changelog.md#ff7-steam as an example :) thank you!

**Review by julianxhokaxhiu — CHANGES_REQUESTED** [review]
One last change sorry, spotted it now. Thanks!

File: Changelog.md (hunk 2 above)
Comment (maciej-trebacz) [reply Changelog.md]: @julianxhokaxhiu Done.

File: Changelog.md [hunk 3]
```diff
@@ -46,6 +45,10 @@
 - Voice: Enable worldmap voice acting
 - Voice: Enable tutorial voice acting
 
+## FF7 Steam
+
+- Common: Fix a softlock when trying to change in-game Controller setting to "Normal" (by disabling this option)
+
```
Comment (julianxhokaxhiu) [inline Changelog.md:51]: Thanks! Sorry maybe it was not clear on the previous comment, but can you please put this section after FF7? :)

**Review by julianxhokaxhiu — APPROVED** [review]

**Comment by julianxhokaxhiu** [comment]
Thank you :)

**Pull request merged**